doc/designs/app-passwords.md

```diff
@@ -0,0 +1,312 @@
+# App passwords
+
+**DESIGN STAGE**
+
+## Overview
+
+FreeIPA does not currently allow users to have more than one password.
+Using a single password for the user's account carries potential risk and
+inconvenience with it. If the password is compromised on only one of the devices
+the user uses, the attacker gains full administrative access to the user's
+account and all systems and services associated with it. The user can then halt
+it only by changing this single account's password and reconfiguring all systems
+and applications using that password for authentication (even if only to one
+particular service) to use the new one.
+
+This enhancement therefore brings support for app passwords (sometimes also
+called application-specific passwords), a form of multiple passwords for a single
+user. Only this would, if app passwords are each used only on one device, increase
+convenience in the case that the user can no longer trust a password used on one
+of his devices. Namely, he can revoke or change only that password, without
+needing to revoke or change his other passwords, including the primary password.
+What is more, app passwords will generally not be usable for account management.
+
+As a further possible future enhancement, this design page also outlines a
+proposal to add support for use of app passwords with FreeIPA's Kerberos
+functionality, what would enable to use the HBAC (host-based access control) to
+allow specifying what app passwords can be used to log into a particular system
+or service. The use of HBAC with app passwords would add another security
+benefit on top of those mentioned above - the intruder's access to the user's
+account might be denied not only to the account management, but also to at least
+some of the systems and services that it is used for.
+
+Resolves: https://pagure.io/freeipa/issue/4510
+
+How Google does it: https://support.google.com/accounts/answer/185833?hl=en
+
+## Use cases
+
+### Use with password-only authentication
+
+The user can set multiple passwords per account. Each password can be then used
+for authenticating on only one device. Various passwords can as well be used to
+authenticate to various services or their sets. These divisions can even be
+combined, thus - in the most extreme case - having one password for
+authenticating to one specific service on one particular device. Granularity of
+such divisions will be fully up to user's discretion.
```

Comment:
I think the user case needs to be expanded a bit. It mentions devices but there are no examples of these devices, similarly services. Would a more concrete example be something like "Allow a user to authenticate to a specific web application using a separate password"? Or allow, for example, a user to have a separate password for system login (their "main" account password) and for checking mail?

As best I can understand these app account entries are non-POSIX so won't allow system logins.

Reply:
Indeed, as written in the design, these app passwords will not be usable for account management, that means logging in directly into FreeIPA. I will make it more clear in the design what 'devices', 'systems' and 'services' mean.